docs/70-indexing/1-create-compound-index.mdx

```diff
@@ -81,44 +82,64 @@ In this exercise, you will build a compound index following the ESR rule, compar
-
-    // ...
-    // Add the CompoundIndex annotation
-    @CompoundIndex(name = "user_returned_borrow_idx", def = "{'user._id': 1, 'returnedDate': 1, 'borrowDate': 1}")
-    public class IssueDetail {
-        // ...
-
-        // Constructors, Getters, Setters
-    }
+1. Let’s start with our [`java-server/src/main/java/com/mongodb/devrel/library/domain/model/IssueDetail.java`](https://github.com/mongodb-developer/library-management-system/blob/java-server/java-server/src/main/java/com/mongodb/devrel/library/domain/model/IssueDetail.java) record. Right now, it looks like this:
+    ```
+    @Document(collection = "issueDetails")
+    public record IssueDetail(
+        // fields ..
+    ) {}
+    ``` 
+
+2. Now let’s optimize queries that need to filter or sort by multiple fields. For example, we often query by user._id, returnedDate, and borrowDate together.
+    To speed this up, we can add a compound index using the **@CompoundIndex** annotation.  
+    ```java  
+    @Document(collection = "issueDetails")
+    @CompoundIndex(
+            name = "user_returned_borrow_idx", 
+            def = "{'user._id': 1, 'returnedDate': 1, 'borrowDate': 1}")
+    public record IssueDetail(
+        // fields
+    ){}
     ```
 
     :::info
     The index is created on the `issueDetails` collection to support the query that finds the issue details for a user with a specific `userID`, where the `borrowDate` is greater than or equal to a specific `date`, and sorts the results by `returnedDate` in descending order.
 
     This is compound index and it follows the ESR rule: Equality, Sort, and Range. This ensures optimal performance for the query.
     :::
-1. Add to `application.properties` this line to enable automatic creation of indexes
+3. Open the `application.yml` and include the property `auto-index-creation: true`. At the end, your configuration will look like this:
 
+
+    ```yaml
+    spring:
+      data:
+        mongodb:
+          uri: ${MONGODB_URI}
+          database: library
+          auto-index-creation: true
     ```
-    spring.data.mongodb.auto-index-creation=true
-    ```
+
+This ensures that any indexes defined in your domain models (for example with @Indexed) will be created automatically by Spring Data MongoDB at startup.
+
 1. Stop the running app.
     1. Locate the bottom panel and click on the `TERMINAL` tab.
     1. Press Ctrl+C to interrup the running app
 
 1. Restart the app typing in the Terminal:
 
     ```bash
-    mvn spring-boot:start
+    mvn spring-boot:run
     ```
+As soon as the application starts, you will see log entries showing the creation of indexes, similar to the image below.
+
+<Screenshot url="https://github.com/mongodb-developer/library-management-system" src="img/screenshots/70-indexing/index-creation.png" alt="Spring Index creation" />
+
+To double-check, you can either open **MongoDB Compass** and inspect the Indexes tab of the issueDetails collection,
```

Comment:
```suggestion
To double-check, you can either open **MongoDB Compass** and inspect the Indexes tab of the `issueDetails` collection,
```

Reply:
We haven't discussed Compass at this point. Is there a way to do this from the application? Maybe through the MongoDB Extension for VS Code?